### PR TITLE
feat: Set user agent for http client

### LINKED ISSUE #2900: User agent fetch and import
[It looks like deno makes requests with a user agent of "reqwest/...", which seems to be the default coming from that Rust package.](https://github.com/pikapkg/web/issues/97#issuecomment-529621401) 
Maybe we need a standard compliant and Deno identified user agent when we import and run `deno fetch` with a url.

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes #2900

### CODE REVIEW

**Comment by kitsonk** [comment]
Test?

**Comment by bartlomieju** [comment]
> Test?

I'm writing it now 😂

File: cli/http_util.rs [hunk 1]
```diff
@@ -1,20 +1,29 @@
...
 use url::Url;
 
 /// Create new instance of async reqwest::Client. This client supports
 /// proxies and doesn't follow redirects.
 pub fn get_client() -> Client {
+  let mut headers = HeaderMap::new();
+  headers.insert(
+    USER_AGENT,
+    format!("Deno/{}", version::DENO).parse().unwrap(),
+  );
```
Comment (ry) [inline cli/http_util.rs]: This prevents reqwest from sending its own user agent? did you check that there are not two user agent header lines?

File: js/fetch_test.ts [hunk 2]
```diff
@@ -220,6 +220,16 @@ testPerm({ net: true }, async function fetchInitBlobBody(): Promise<void> {
   assert(response.headers.get("content-type").startsWith("text/javascript"));
 });
 
+testPerm({ net: true }, async function fetchUserAgent(): Promise<void> {
+  const data = "Hello World";
+  const response = await fetch("http://localhost:4545/echo_server", {
+    method: "POST",
+    body: new TextEncoder().encode(data)
```
Comment (ry) [inline js/fetch_test.ts]: is a post request necessary?

File: cli/http_util.rs (hunk 1 above)
Comment (bartlomieju) [reply cli/http_util.rs]: Yes, I dumped headers in echo server:
```
'user-agent: Deno/0.17.0\r\naccept: */*\r\naccept-encoding: gzip\r\nhost: localhost:4545\r\ncontent-length: 11\r\n'
```
There's only single `user-agent` header.

**Pull request merged**